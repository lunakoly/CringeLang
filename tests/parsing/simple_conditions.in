if 10 - 5
    sayHello
else
    sayBye

if iAmTrue doThings
else doOtherThings

if True sayTest else
    a = b
    c = 10 + a

if test sayFest
    errorIndent
    error2

a, b = 10, 20