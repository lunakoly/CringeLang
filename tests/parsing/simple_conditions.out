==== Raw AST ====
[if (10 - 5) [[[sayHello]]] else [[[sayBye]]], if iAmTrue [[doThings]] else [[doOtherThings]], if True [[sayTest]] else [[([a] = [b]), ([c] = [(10 + a)])]], if test [[sayFest]], [[errorIndent], [error2]], ([a, b] = [10, 20])]

==== Resolved AST ====
[if (10 - 5) [[[sayHello]]] else [[[sayBye]]], if iAmTrue [[doThings]] else [[doOtherThings]], if True [[sayTest]] else [[([a] = [b]), ([c] = [(10 + a)])]], if test [[sayFest]], [[errorIndent], [error2]], ([a, b] = [10, 20])]

==== Diagnostics ====
 13 | ...<newline>    errorIndent...
                  ~~~~^~~~~~~~~~~
Error > Unexpected indent level > `INDENT` shouldn't go here.

==== Done ====
